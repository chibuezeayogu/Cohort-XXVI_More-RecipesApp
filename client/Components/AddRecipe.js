import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;