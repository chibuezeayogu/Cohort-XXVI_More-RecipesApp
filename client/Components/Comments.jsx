import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
