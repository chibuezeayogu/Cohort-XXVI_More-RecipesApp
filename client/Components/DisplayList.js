import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;