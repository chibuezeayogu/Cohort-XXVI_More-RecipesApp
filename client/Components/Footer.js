import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;