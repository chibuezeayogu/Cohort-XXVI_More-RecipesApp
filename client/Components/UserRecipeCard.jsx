import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import moment from 'moment';
import swal from 'sweetalert';

/**
 *
 * @description displays each recipe
 *
 * @method
 *
 * @param {Object} props - property object
 *
 *  @returns {undefined}
 */
export class UserRecipeCard extends Component {
  handelEditRecipe(id) {
    this.props.history.push(`/user/recipes/${id}/edit`);
  }

  handelDeleteRecipe(id, title) {
    swal({
      title: 'Are you sure?',
      text: `You want to delete "${title}"!`,
      icon: 'warning',
      buttons: true,
      dangerMode: true,
    })
      .then((willDelete) => {
        if (willDelete) {
          this.props.delRecipe(id);
        }
      });
  }


  render() {
    const { recipe } = this.props;
    return (
      <div className="col s12 m6 l4">
        <div className="card hoverable">
          <Link
            to={`/recipes/${recipe.id}`}
          >
            <img
              src={this.props.recipe.imageUrl}
              alt=""
              className="responsive-img img-height"
              style={{ width: '100%' }}
            />
          </Link>
          <div
            className="card-content black-text grey lighten-5"
          >
            <span className="card-title text-title truncate">
              {recipe.title}
            </span>
            <p className="text-description truncate">
              {recipe.description}
            </p>
            <hr />
            <p className="text-description truncate">
              Posted { moment(new Date(recipe.createdAt)).fromNow()}
            </p>
          </div>
          <div
            className="card-action black-text grey lighten-4 center"
          >
            <a className="black-text">
              <i
                className="fa fa-thumbs-o-up"
                aria-hidden="true"
              > {recipe.upvotes}
              </i>
            </a>
            <a
              className="black-text"
            >
              <i
                className="fa fa-thumbs-o-down"
                aria-hidden="true"
              > {recipe.downvotes}
              </i>
            </a>
            <a
              className="black-text"
              onClick={() => this.handelEditRecipe(this.props.recipe.id)}
              style={{ cursor: 'pointer' }}
            >
              <i
                  className="fa fa-pencil-square-o black-text"
                  aria-hidden="true"
                />
            </a>

            <a
              className="black-text"
              onClick={() => this.handelDeleteRecipe(this.props.recipe.id, this.props.recipe.title)}
              style={{ cursor: 'pointer' }}
            >
              <i
                className="fa fa-trash"
                aria-hidden="true"
              />
            </a>
          </div>
        </div>
      </div>
    );
  }
}

UserRecipeCard.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number.isRequired,
    upvotes: PropTypes.number.isRequired,
    downvotes: PropTypes.number.isRequired,
    views: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired
  }).isRequired
};

export default UserRecipeCard;
