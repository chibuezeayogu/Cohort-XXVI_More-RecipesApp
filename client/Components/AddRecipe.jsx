require('dotenv').config();

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { createRecipe } from '../action/actionCreators';
import SmallPreloader from './SmallPreloader.jsx';
import Footer from './Footer/Footer.jsx';
import UserMenu from './Header/UserMenu.jsx';
import { validateAddRecipe } from '../middleware/validateInputs';

const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;

/**
 * @description AddRecipe component for user to create a new recipe
 *
 * @class
 *
 * @extends Component
 *
 */
class AddRecipe extends Component {
   /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      ingredients: '',
      procedures: '',
      imageUrl: '',
      image: {},
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
  }

   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/signin');
    }
  }
   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Object} nextProps
   *
   * @returns {void}
   *
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.recipeReducer.isCreated) {
      this.props.history.push('/recipes');
    }
  }


  handleChange(e) {
    e.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * @description get image input and add to state
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   */
  handleImageChange(e) {
    e.preventDefault();
    const file = e.target.files[0];
    this.setState({ errors: {}, image: {} });
    this.setState({ image: file });
  }

  /**
   *
   * @description submits form
   *
   * @method
   *
   * @memberOf AddRecipe
   *
   * @param {Event} e
   *
   * @returns {void}
   *
   */
  handleOnsubmit(e) {
    e.preventDefault();
    const err = validateAddRecipe(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }

    this.setState({ disabled: true });
    const formData = new FormData();
    formData.append('file', this.state.image);
    formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);

    axios({
      url: CLOUDINARY_URL,
      method: 'POST',
      data: formData,
    }).then((data) => {
      this.setState({ imageUrl: data.data.secure_url, disabled: false });
      const {
        title, description, ingredients, procedures, imageUrl,
      } = this.state;
      console.log(imageUrl, 'image');
      this.props.createRecipe(title, description, ingredients, procedures, imageUrl);
    }).catch(() => { 
      this.setState({ disabled: false });
      Materialize.toast("Error! Please try again", 4000, 'red');
    });
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @returns {void}
   *
   */
  render() {
    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="addrecipe-form z-depth-4">
              <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Add Recipe</h4>
                <hr />
              </div>
              <form onSubmit={this.handleOnsubmit}>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value={this.state.title}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="title" className="active">Title</label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.titleError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="description"
                      name="description"
                      type="text"
                      value={this.state.description}
                      onChange={this.handleChange}
                    />
                    <label
                      htmlFor="description"
                      className="active"
                    >Description
                    </label>
                  </div>
                  <span className="right red-text">
                    {this.state.errors.descriptionError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="ingredients"
                      name="ingredients"
                      value={this.state.ingredients}
                      onChange={this.handleChange}
                      placeholder="enter ingredients seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.ingredientsError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="procedures"
                      name="procedures"
                      value={this.state.password}
                      onChange={this.handleChange}
                      placeholder="enter procedures seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.proceduresError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <i className="material-icons prefix">insert_photo</i>
                    <input
                      type="file"
                      name="profileImage"
                      onChange={e => this.handleImageChange(e)}
                    />
                  </div>
                  <span className="right red-text">
                    {this.state.errors.imageError}
                  </span>
                </div>
                <div className="row">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                  >Post
                  </button>
                </div>
                <hr />
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

AddRecipe.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  recipeReducer: PropTypes.shape({
    isCreated: PropTypes.bool.isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  recipeReducer: state.recipeReducer,
});

export default connect(mapStateToProps, { createRecipe })(AddRecipe);

