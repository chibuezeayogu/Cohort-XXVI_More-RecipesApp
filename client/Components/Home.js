import React, { Component } from 'react';
import { Row, Input, Icon, Textarea, Slider, Slide, Parallax } from 'react-materialize';
import NavBar from  './NavBar';
import Footer from  './Footer';


class Home extends Component {
    render() {
        return (
            <div>
                <Slider className="full-screen slide-margin-top">
                    <Slide
                        src="https://static.pexels.com/photos/161640/abstract-barbecue-barbeque-bbq-161640.jpeg"
                        title="More Recipes">
                        More-Recipes provides a platform for users to share the awesome and exciting recipe ideas they have invented or learnt.
                    </Slide>
                    <Slide
                        src="https://static.pexels.com/photos/213621/pexels-photo-213621.jpeg"
                        title="Easy to Setup"
                        placement="left">
                        npm install react-materialize.
                    </Slide>
                    <Slide
                        src="https://static.pexels.com/photos/51423/bun-chocolate-sweets-dessert-51423.jpeg"
                        title="Unlimited Oppurtunities"
                        placement="right">
                        Contribute and Help to Grow this Project.
                    </Slide>
                </Slider>
            </div>
        );
    }
}

export default Home;