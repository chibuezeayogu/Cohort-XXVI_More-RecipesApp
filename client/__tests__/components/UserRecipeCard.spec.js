import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { UserRecipeCard } from '../../Components/UserRecipeCard';
import mockData from '../__mock__/actionMockData';

const props = {
  recipe: mockData.recipe,
  history: {
    push: jest.fn()
  }
};

describe('<UserRecipeCard />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UserRecipeCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('handelEditRecipe function', () => {
  it('should navigate to the edit recipe page', () => {
    const wrapper = shallow(<UserRecipeCard {...props} />);
    const handelEditRecipeSpy = jest
      .spyOn(wrapper.instance(), 'handelEditRecipe');
    wrapper.instance().handelEditRecipe();
    expect(handelEditRecipeSpy).toHaveBeenCalled();
    expect(wrapper.instance().props.history.push).toHaveBeenCalled();
  });
});

describe('handelDeleteRecipe function', () => {
  it('should delete a recipe when clicked', () => {
    const wrapper = shallow(<UserRecipeCard {...props} />);
    const handelDeleteRecipeSpy = jest
      .spyOn(wrapper.instance(), 'handelDeleteRecipe');
    wrapper.instance().handelDeleteRecipe();
    expect(handelDeleteRecipeSpy).toHaveBeenCalled();
  });
});
