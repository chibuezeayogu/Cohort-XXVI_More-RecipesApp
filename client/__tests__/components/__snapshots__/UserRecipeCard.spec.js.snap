// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRecipeCard /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="col s12 m6 l4"
  >
    <div
      className="card hoverable"
    >
      <Link
        replace={false}
        to="/recipes/undefined"
      >
        <img
          alt=""
          className="responsive-img img-height"
          src={undefined}
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </Link>
      <div
        className="card-content black-text grey lighten-5"
      >
        <span
          className="card-title text-title truncate"
        />
        <p
          className="text-description truncate"
        />
        <hr />
        <p
          className="text-description truncate"
        >
          Posted 
          Invalid date
        </p>
      </div>
      <div
        className="card-action black-text grey lighten-4 center"
      >
        <a
          className="black-text"
        >
          <i
            aria-hidden="true"
            className="fa fa-thumbs-o-up"
          >
             
          </i>
        </a>
        <a
          className="black-text"
        >
          <i
            aria-hidden="true"
            className="fa fa-thumbs-o-down"
          >
             
          </i>
        </a>
        <a
          className="black-text"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <i
            aria-hidden="true"
            className="fa fa-pencil-square-o black-text"
          />
        </a>
        <a
          className="black-text"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <i
            aria-hidden="true"
            className="fa fa-trash"
          />
        </a>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="col s12 m6 l4"
    >
      <div
        className="card hoverable"
      >
        <Link
          replace={false}
          to="/recipes/undefined"
        >
          <img
            alt=""
            className="responsive-img img-height"
            src={undefined}
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </Link>
        <div
          className="card-content black-text grey lighten-5"
        >
          <span
            className="card-title text-title truncate"
          />
          <p
            className="text-description truncate"
          />
          <hr />
          <p
            className="text-description truncate"
          >
            Posted 
            Invalid date
          </p>
        </div>
        <div
          className="card-action black-text grey lighten-4 center"
        >
          <a
            className="black-text"
          >
            <i
              aria-hidden="true"
              className="fa fa-thumbs-o-up"
            >
               
            </i>
          </a>
          <a
            className="black-text"
          >
            <i
              aria-hidden="true"
              className="fa fa-thumbs-o-down"
            >
               
            </i>
          </a>
          <a
            className="black-text"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              aria-hidden="true"
              className="fa fa-pencil-square-o black-text"
            />
          </a>
          <a
            className="black-text"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              aria-hidden="true"
              className="fa fa-trash"
            />
          </a>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserRecipeCard
        history={
          Object {
            "push": [MockFunction],
          }
        }
        recipe={
          Array [
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-28T22:46:03.051Z",
              "description": "asdfafd",
              "downvotes": 0,
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "adfadfadf; adfadfadf",
              "procedures": "adfafadf; adfadfadf",
              "title": "CHANGED",
              "updatedAt": "2018-02-06T19:28:36.149Z",
              "upvotes": 1,
              "views": 0,
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserRecipeCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "history": Object {
            "push": [MockFunction],
          },
          "recipe": Array [
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-28T22:46:03.051Z",
              "description": "asdfafd",
              "downvotes": 0,
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "adfadfadf; adfadfadf",
              "procedures": "adfafadf; adfadfadf",
              "title": "CHANGED",
              "updatedAt": "2018-02-06T19:28:36.149Z",
              "upvotes": 1,
              "views": 0,
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="col s12 m6 l4"
        >
          <div
            className="card hoverable"
          >
            <Link
              replace={false}
              to="/recipes/undefined"
            >
              <img
                alt=""
                className="responsive-img img-height"
                src={undefined}
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
            </Link>
            <div
              className="card-content black-text grey lighten-5"
            >
              <span
                className="card-title text-title truncate"
              />
              <p
                className="text-description truncate"
              />
              <hr />
              <p
                className="text-description truncate"
              >
                Posted 
                Invalid date
              </p>
            </div>
            <div
              className="card-action black-text grey lighten-4 center"
            >
              <a
                className="black-text"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                </i>
              </a>
              <a
                className="black-text"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                </i>
              </a>
              <a
                className="black-text"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="fa fa-pencil-square-o black-text"
                />
              </a>
              <a
                className="black-text"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="fa fa-trash"
                />
              </a>
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="col s12 m6 l4"
        >
          <div
            className="card hoverable"
          >
            <Link
              replace={false}
              to="/recipes/undefined"
            >
              <img
                alt=""
                className="responsive-img img-height"
                src={undefined}
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
            </Link>
            <div
              className="card-content black-text grey lighten-5"
            >
              <span
                className="card-title text-title truncate"
              />
              <p
                className="text-description truncate"
              />
              <hr />
              <p
                className="text-description truncate"
              >
                Posted 
                Invalid date
              </p>
            </div>
            <div
              className="card-action black-text grey lighten-4 center"
            >
              <a
                className="black-text"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-up"
                >
                   
                </i>
              </a>
              <a
                className="black-text"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                   
                </i>
              </a>
              <a
                className="black-text"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="fa fa-pencil-square-o black-text"
                />
              </a>
              <a
                className="black-text"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="fa fa-trash"
                />
              </a>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserRecipeCard
    history={
      Object {
        "push": [MockFunction],
      }
    }
    recipe={
      Array [
        Object {
          "addedBy": 1,
          "createdAt": "2018-01-28T22:46:03.051Z",
          "description": "asdfafd",
          "downvotes": 0,
          "id": 1,
          "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
          "ingredients": "adfadfadf; adfadfadf",
          "procedures": "adfafadf; adfadfadf",
          "title": "CHANGED",
          "updatedAt": "2018-02-06T19:28:36.149Z",
          "upvotes": 1,
          "views": 0,
        },
      ]
    }
  />,
}
`;
